### PR TITLE
add err handling and change some order

### PR DESCRIPTION
**Pull request by allencloud**
1. add err handling when constructing a http request
2. check if request calling succeeds first, then parse response
 
Signed-off-by: allencloud <allen.sun@daocloud.io>

### CODE REVIEW

File: client/request.go [hunk 1]
```diff
@@ -110,13 +110,20 @@ func (cli *Client) sendClientRequest(ctx context.Context, method, path string, q
...
 		}
 
 		return serverResp, fmt.Errorf("An error occurred trying to connect: %v", err)
 	}
 
+	if resp != nil {
+		serverResp.statusCode = resp.StatusCode
+	}
+
+	defer resp.Body.Close()
```
Comment (allencloud) [inline client/request.go]: I wonder if this is a good way. When Close the resp.Body, serverResp.body will be influenced. 
And I think it is not a good way to define body in serverResponse as an  io.ReadCloser
```
type serverResponse struct {
	body       io.ReadCloser
	header     http.Header
	statusCode int
}
```
Usually we get the http response, get the body content using `ioutil.ReadAll` or other func, the Close the resp.Body.

Comment (calavera) [inline client/request.go]: I don't think we should close the body here. Functions that stream responses depend on it to not be closed.

**Comment by allencloud** [comment]
@calavera 
Removed that part already, and update the pr name and description.

**Comment by calavera** [comment]
LGTM

**Comment by allencloud** [comment]
Any update ?

**Comment by vdemeester** [comment]
LGTM 🐡

**Pull request merged**